homeassistant/components/nightscout/sensor.py

```diff
@@ -120,3 +136,43 @@ def _parse_icon(self) -> str:
     def extra_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
+
+
+class Battery(CoordinatorEntity, SensorEntity):
+    """Battery sensor of Nightscout device."""
+
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+
+    def __init__(self, coordinator, name, unique_id):
+        """Initialize the Nightscout sensor."""
+        super().__init__(coordinator)
+        self._attr_unique_id = unique_id
+        self._attr_name = name
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        uploader = self.coordinator.data[self._attr_name].uploader
+        info = {
+            "identifiers": {(DOMAIN, self.unique_id)},
+        }
+        if hasattr(uploader, "type"):
+            info["model"] = uploader.type
+
+        return info
+
+    @property
+    def available(self):
+        """Return if the sensor data are available."""
+        return (
+            hasattr(self.coordinator.data[self._attr_name], "uploader")
```

Comment:
It's a bit weird that the `uploader` attribute would be missing. Attributes should always be present after instantiation.

Reply:
Yeah, so... a nightscout device can include a `uploader` object or not, depending on which device it is. Their API is a little confusing and not very well documented.

You can check out [the documentation here](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/nightscout/cgm-remote-monitor/master/lib/server/swagger.yaml#/Devicestatus/get_devicestatus_). `uploader` `pump` and `xdripjs` seem to be mutually exclusive and a DeviceStatus could have any of them.

Reply:
Normally a Python instance sets optional attributes to None.